### ISSUE TITLE
Update the part in the documentation comment about array_filter

### ISSUE THREAD

**Issue by avpaderno** [backdrop/backdrop-issues#6303]
[`system_settings_form_submit()`](https://docs.backdropcms.org/api/backdrop/core%21modules%21system%21system.module/function/system_settings_form_submit/1) has the following documentation comment.

```php
/**
 * Form submission handler for system_settings_form().
 *
 * If you want node type configure style handling of your checkboxes,
 * add an array_filter value to your form.
 *
 * @since 1.6.0 Restored and made compatible with configuration management.
 */
```

The part about the _array_filter_ value is true only for forms that do not define `$form['#config']` and when the Drupal compatibility layer is enabled, since that function uses the following code. (The comments in the code are mine.)

```php
  if (!empty($form['#config'])) {
    $configs = _system_sort_form_values_by_config($form, $form_state, $form['#config']);
    foreach ($configs as $bucket => $data) {
      config_set_multiple($bucket, $data);
    }
  }
  else {
    watchdog('form', 'Backdrop 2.0 and later will require forms using system_settings_form() to set $form[\'#config\'].', array(), WATCHDOG_DEPRECATED);
    if (!settings_get('backdrop_drupal_compatibility')) {
      backdrop_set_message(t('Settings not saved: Drupal compatibility layer is disabled and this form has not been updated to use Backdrop configuration management.'), 'error');
      return;
    }

    // The form does not set a value for $form['#config'] and the Drupal compatibility layer
    // is enabled.
    foreach ($form_state['values'] as $key => $value) {
      if (is_array($value) && isset($form_state['values']['array_filter'])) {
        $value = array_keys(array_filter($value));
      }
      variable_set($key, $value);
    }
  }
```

Either that documentation comment is changed to make clear `'array_filter'` is used when the Drupal compatibility layer is enabled, and the form does not set `$form['#config']`, or the code is changed to handle `'array_filter'` in all the cases.

IMO, `'array_filter'` should only be used for BC with Drupal 7. I would rather change the documentation comment to reflect that.

**Comment by kiamlaluno** [comment]
I would rather remove the *If you want node type configure style handling of your checkboxes, add an array_filter value to your form.* sentence from the documentation comment for the submission handler and add a comment to the code that handles `$form['array_filter']`.

**Comment by kiamlaluno** [comment]
I will create a PR in the next 24 hours.

**Comment by yorkshire-pudding** [comment]
Hi @avpaderno - thanks for raising this issue and the PR. I was baffled by this function doc comment and found this issue.
I've suggested a small expansion of the explanation in the code comment.

**Comment by avpaderno** [comment]
@yorkshire-pudding Thank you! I committed the suggested change.

**Comment by yorkshire-pudding** [comment]
Thanks @avpaderno - there is an unrelated cspell failure.  RTBC

**Comment by argiepiano** [comment]
Good catch. We may also want to get rid of `$form['array_filter']` in the [book module](https://github.com/backdrop/backdrop/blob/003ccb38a84d7b1c6181d60d439524c9cf4d71be/core/modules/book/book.admin.inc#L106), which doesn't do anything. That form doesn't even use `system_settings_form()`.

**Comment by yorkshire-pudding** [comment]
> Good catch. We may also want to get rid of `$form['array_filter']` in the [book module](https://github.com/backdrop/backdrop/blob/003ccb38a84d7b1c6181d60d439524c9cf4d71be/core/modules/book/book.admin.inc#L106), which doesn't do anything. That form doesn't even use `system_settings_form()`.

@argiepiano - @avpaderno has already created an issue for that:
https://github.com/backdrop/backdrop-issues/issues/6304

**Comment by quicksketch** [comment]
Thank you @avpaderno, @yorkshire-pudding, and @argiepiano! The PR at https://github.com/backdrop/backdrop/pull/4589 has been merged into 1.x and 1.31.x.
